snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/SnomedRf2ExportRequest.java

```diff
@@ -15,273 +15,777 @@
+		final Map<String, Branch> versionBranchesByName = FluentIterable.from(versionBranches)
+				.uniqueIndex(b -> b.name());
+
+		final Branch cutoffBranch = getBranch(cutoffPath);
+		final long cutoffBaseTimestamp = getCutoffBaseTimestamp(cutoffBranch, versionParentPath);
+
+		// Remove all code system versions which were created after the cut-off date, or don't have a corresponding branch 
+		candidates.removeIf(v -> !versionBranchesByName.containsKey(v.getVersionId())
+				|| versionBranchesByName.get(v.getVersionId()).baseTimestamp() > cutoffBaseTimestamp);
+
+		versionsToExport.addAll(candidates);
+
+		// Exit early if only an extension code system should be exported, or we are already at the "base" code system
+		if (extensionOnly || Strings.isNullOrEmpty(codeSystemEntry.getExtensionOf())) {
+			return;
+		}
+
+		// Otherwise, collect applicable versions using this code system's working path
+		final CodeSystemEntry extensionEnty = getCodeSystem(codeSystemEntry.getExtensionOf());
+		collectVersionsToExport(versionsToExport, extensionEnty, codeSystemEntry.getBranchPath());
+	}
+
+	private void filterVersions(final TreeSet<CodeSystemVersionEntry> versionsToExport) {
+		// Is there anything to filter?
+		if (versionsToExport.isEmpty()) {
+			return;
+		}
+
+		switch (releaseType) {
+			case DELTA:
+				// Delta export: keep versions that fall into the effective time range
+				versionsToExport.removeIf(v -> {
+					return false
+							|| (startEffectiveTime != null && v.getEffectiveDate() < startEffectiveTime.getTime())
+							|| (endEffectiveTime != null && v.getEffectiveDate() > endEffectiveTime.getTime());
+				});
+				break;
+			case FULL:
+				// Full export: nothing to remove, all versions should be exported
+				break;
+			case SNAPSHOT:
+				// Snapshot export: keep only the latest visible version
+				versionsToExport.retainAll(ImmutableSet.of(versionsToExport.last()));
+				break;
+			default:
+				throw new IllegalStateException("Unexpected RF2 release type '" + releaseType + "'.");
+		}
+	}
+
+	private Path createExportDirectory(final UUID exportId) {
 		try {
-			SnomedRf2ExportModel model = toExportModel(context);
-			final File file = doExport(model, context.service(IProgressMonitor.class));
-			if (file == null || model.getExportResult().getResult() == Result.EXCEPTION) {
-				throw new SnowowlRuntimeException(model.getExportResult().getMessage());
+			return Files.createTempDirectory("export-" + exportId + "-");
+		} catch (final IOException e) {
+			throw new SnowowlRuntimeException("Failed to create working directory for export.", e);
+		}
+	}
+
+	private Path createReleaseDirectory(final Path exportDirectory, final String latestEffectiveTime) {
+		final String releaseStatus = includePreReleaseContent
+				? "BETA"
+						: "PRODUCTION";
+
+		final Path releaseDirectory = exportDirectory.resolve(String.format("SNOMEDCT_RF2_%s_%sT120000Z", releaseStatus, latestEffectiveTime));
```

Comment:
⚠️ Hardcoded hours!

Reply:
The release root directory name is computed with the transient effective date in this case, which does not include time-level information.